SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -1,17 +1,49 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 
 #include "SpatialCommandUtils.h"
+#include "SpatialGDKServicesModule.h"
 
 DEFINE_LOG_CATEGORY(LogSpatialCommandUtils);
 
-bool SpatialCommandUtils::AttemptSpatialAuth(bool IsRunningInChina)
+namespace
+{
+	FString ChinaEnvironmentArgument = TEXT(" --environment=cn-production");
+}
+
+void SpatialCommandUtils::ExecuteSpatialCommandAndReadOutput(FString Arguments, const FString& DirectoryToRun, FString& OutResult, int32& OutExitCode, bool bIsRunningInChina)
+{
+	if (bIsRunningInChina)
+	{
+		Arguments += ChinaEnvironmentArgument;
```

Comment:
Can this not be added before this function is called? Then we don't need to copy the entire argument list here.

Reply:
Then we can also get rid of this function presumably.

Reply:
I don't follow - Isn't the point of this function to ensure we wrap ALL spatial calls with the China arg when if we are in China? Are you suggesting we manually check and append in every call site?

Reply:
^ 
I talked with Sami and he likes the sentiment of the change he just didn't like the incompleteness of the change. I will be making a follow up PR to make this beautiful.